src/coreclr/src/vm/jitinterface.cpp

```diff
@@ -12865,6 +12894,9 @@ PCODE UnsafeJitFunction(PrepareCodeConfig* config,
                                                   nativeCodeVersion);
             LOG((LF_CORDB, LL_EVERYTHING, "Got through CallCompile MethodWithSEHWrapper\n"));
 
+            g_cbILJitted += methodInfo.ILCodeSize;
```

Comment:
These counters are subject to few sources of error. Most of them probably aren't worth the perf overhead to address, but lets at least document them in comments. If we get complaints that inprecise measurements are causing trouble in the future we'll know potential trouble spots and can revisit our decisions about what is worth addressing.

1. We aren't using InterlockedIncrement so it is possible that two or more threads will race to update the values and some updates may be lost.
2. The JIT request may not succeed and will try again (the retry loop starts at 12770). We are currently counting attempts, not successes. I'd recommend moving the count down adjacent to:
`````
LOG((LF_JIT, LL_INFO10000,
            "Jitted Entry at"
`````
3. UnsafeJitFunction in rare cases can race to JIT the same method on multiple threads, and later one of the jittings will be discarded. The counter will treat this as 2 jittings whereas most of our other telemetry treats it as 1 jitting.

It would be useful to do some testing on a good sized app to ensure this counter typically matches the number jit events reported by ETW to ensure we didn't miss other error sources or underestimate the ones listed here.

Reply:
Thanks Noah! I just modified it to use FastInterlockedIncrement - I wasn't aware that UnsafeJitFunction could get called from multiple threads. Would it be too much perf overhead to do this? I think it should be fine since a lot of other counters that gets incremented often also use FastInterlockedIncrement internally (i.e. thread contention counts) but let me know if this sounds like a bad idea. 

I'll try to also find a larger sized app to test this against ETW statistics and see how much we deviate from what we currently report via ETW.

Reply:
My understanding is that InterlockedIncrement costs 10s of CPU cycles per call though whether its low 10s or high 10s or my data is unreliable I can't really say. Every estimate I've seen is generally littered in caveats that makes me think the actual cost is highly scenario and hardware dependent. Ultimately its probably splitting hairs, for a typical app this will probably only add a small fraction of a millisecond to startup, and on the flip side the error case is likely to be both rare and small. I'd guess its the kind of thing that regardless which way we decide we are unlikely to hear about it again. So my suggestion is leave it however you have it : )